extmod/modurandom.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016 Paul Sokolovsky
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <assert.h>
+#include <string.h>
+
+//#include "py/nlr.h"
+#include "py/runtime.h"
+
+#if MICROPY_PY_URANDOM
+
+// Yasmarang random number generator
+// by Ilya Levin
+// http://www.literatecode.com/yasmarang
+// Public Domain
+
+STATIC uint32_t pad = 0xeda4baba, n = 69, d = 233;
+STATIC uint8_t dat = 0;
+
+uint32_t yasmarang(void)
+{
+   pad += dat + d * n;
+   pad = (pad<<3) + (pad>>29);
+   n = pad | 2;
+   d ^= (pad<<31) + (pad>>1);
+   dat ^= (char) pad ^ (d>>8) ^ 1;
+
+   return (pad^(d<<5)^(pad>>18)^(dat<<1));
+}  /* yasmarang */
+
+// End of Yasmarang
+
+STATIC mp_obj_t mod_urandom_getrandbits(mp_obj_t num_in) {
+    int n = MP_OBJ_SMALL_INT_VALUE(num_in);
```

Comment:
Why not mp_obj_get_int?  Using that saves potential issues where people make small numbers from a bignum (eg 1234567890123456 & 0xff).  That's not common, but if you hit that issue then it's really strange and hard to work out what is happening.